salt/minion.py

```diff
@@ -1449,6 +1449,20 @@ def get_executor(name):
             ret,
             timeout=minion_instance._return_retry_timer()
         )
+
+        # Add default returners from minion config
+        if isinstance(opts.get('return'), six.string_types):
```

Comment:
We should move the return data type check into where the minion config is generated, that way we on't do the type check and transform one time. What do you think @cachedout ?